### PR TITLE
Fixing config bug and increasing test coverage

### PR DESCRIPTION
**Pull request by SpencerKaiser**
- Added better handling for unknown keys
- Fixing config issues and test coverage

## Pre-Requisites
- [X] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [X] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [X] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Fixing issues with config retrieval and fixing tests

### CODE REVIEW

File: src/entities/config.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export class Config extends BaseEntity {
   /**
    * @deprecated Use `Config.getValueAs(key, 'boolean', true)` instead
    */
+  /* istanbul ignore next */
```
Comment (johnkahn) [inline src/entities/config.ts:27]: Does this method still get the deprecated warning with this comment between it?

**Review by johnkahn — CHANGES_REQUESTED** [review]

File: src/tests/entities/config.test.ts [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+    configFindOneSpy.mockResolvedValueOnce(null);
+    await expect(Config.getValueAs('adminSecret', 'string', true)).rejects.toThrow();
   });
 
-  it('a non-existent toggle will be false', async () => {
-    const toggleValue = await Config.findToggleForKey('non-existent toggle key' as KnownConfig);
-    expect(toggleValue).toEqual(false);
+  it('will return null for an item that cannot be found if throw is disabled', async () => {
+    configFindOneSpy.mockResolvedValueOnce(null);
+    expect(await Config.getValueAs('adminSecret', 'string', false)).toEqual(null);
```
Comment (johnkahn) [inline src/tests/entities/config.test.ts:18]: To match the rest of your other tests (same with the other resolves ones)

```suggestion
    expect(Config.getValueAs('adminSecret', 'string', false)).resolves.toEqual(null);
```

**Review by johnkahn — CHANGES_REQUESTED** [review]

**Comment by johnkahn** [comment]
> Should have been a `Comment` 🙃

True, but can I get an answer on it?

File: src/entities/config.ts (hunk 1 above)
Comment (SpencerKaiser) [reply src/entities/config.ts:27]: Yep, I checked that

**Review by johnkahn — DISMISSED** [review]

**Review by johnkahn — APPROVED** [review]

**Review by Ulfingrad — APPROVED** [review]
Not 100% confident I fully understand the implementation, but as far as I can tell it looks good

**Review by dre8597 — APPROVED** [review]
LGTM, I didn't know expect had an option for checking for throws like this ✅

**Review by dre8597 — APPROVED** [review]

**Pull request merged**